File: Codeforces/576_div2/C.py
def constraint(a,b):
	if(a>b):
		return b
	return 0
n,I = list(map(int,input().split()))
a = list(map(int,input().split()))

K = 1<<(8*I//n)
d=dict()
for i in a:
	d[i]=0
for i in a:
	d[i]+=1
sorted_d = list(d.items())
sorted_d.sort()
num=constraint(len(sorted_d),K)
if(num>0):
	min_c=0
	i=0
	while(i<num):
		min_c+=sorted_d[i][1]
		i+=1
	count=min_c
	total=min_c
	while(i<len(sorted_d)):
		count+=sorted_d[i][1]
		count-=sorted_d[i-num][1]
		total+=sorted_d[i][1]
		if(count>min_c):
			min_c=count
		i+=1
	print(total-min_c)

else:
	print(0)

File: Codeforces/576_div2/C.cpp
#include<bits/stdc++.h>
#define ll long long
using namespace std;
ll constraint(ll a,ll b){
	if(a>b){
		return b;
	}
	return 0;
}
int main(){
	ll n,I;
	cin>>n>>I;
	vector<ll> a(n,0);
	for(ll i=0;i<n;i++){
		cin>>a[i];
	}
	
	//cout<<K<<endl;
	map<ll,ll> m;
	for(ll i=0;i<n;i++){
		m[a[i]]++;
	}
	int q = ceil(log(m.size())/log(2));
	if(n*q<=8*I){
		cout<<"0"<<endl;
		return 0;
	}
	ll K = (ll)1<<(8*I/n);
	ll num = constraint(m.size(),K);
	//cout<<"e"<<endl;
	vector<int> v;
	for(auto x : m){
		v.push_back(x.second);
	}
	if(num>0){
		ll max_c=0;
		ll i = 0;
		while(i<num){
			max_c+= v[i];
			i++;
		}
		ll count=max_c;
		ll total=max_c;
		while(i<v.size()){
			count+= v[i];
			count-= v[i-num];
			total+= v[i];
			if(count>max_c){
				max_c=count;
			}
			i++;
		}
		cout<<total-max_c<<endl;

	}
	else{
		cout<<0<<endl;
	}
return 0;
}